### PR TITLE
A bit more import cleanup work

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

**Comment by cgwalters** [comment]
bot, retest this please

File: src/libostree/ostree-repo-commit.c [hunk 1]
```diff
@@ -3167,6 +3165,249 @@ G_DEFINE_BOXED_TYPE(OstreeRepoCommitModifier, ostree_repo_commit_modifier,
...
+gboolean
+_ostree_repo_import_object (OstreeRepo           *self,
+                            OstreeRepo           *source,
+                            OstreeObjectType      objtype,
+                            const char           *checksum,
+                            OstreeRepoImportFlags flags,
+                            GCancellable         *cancellable,
+                            GError              **error)
+{
+  const gboolean trusted = (flags & _OSTREE_REPO_IMPORT_FLAGS_VERIFY_CHECKSUM) == 0;
```
Comment (jlebon) [inline src/libostree/ostree-repo-commit.c]: Might as well make the flag `_OSTREE_REPO_IMPORT_FLAGS_TRUSTED`. Seems safer to just assume it's untrusted and require the flag to be explicitly set.

File: src/libostree/ostree-repo-commit.c [hunk 2]
```diff
@@ -3167,6 +3165,249 @@ G_DEFINE_BOXED_TYPE(OstreeRepoCommitModifier, ostree_repo_commit_modifier,
...
+           * optimization of hardlinking instead of copying.
+           */
+          *out_was_supported = FALSE;
+          return TRUE;
+        }
+      else
+        return glnx_throw_errno_prefix (error, "linkat");
+    }
+
+  if (objtype == OSTREE_OBJECT_TYPE_COMMIT)
```
Comment (jlebon) [inline src/libostree/ostree-repo-commit.c]: Can't we just hardlink detached metadata too if we succesfully hardlinked the commit?

Comment (cgwalters) [reply src/libostree/ostree-repo-commit.c]: Yeah...we weren't doing that before though, can make that a separate commit?

File: src/libostree/ostree-repo-commit.c (hunk 1 above)
Comment (cgwalters) [reply src/libostree/ostree-repo-commit.c]: Good suggestion, fixup ⬇️

File: src/libostree/ostree-repo-commit.c (hunk 2 above)
Comment (jlebon) [reply src/libostree/ostree-repo-commit.c]: Yup, sure!

**Comment by jlebon** [comment]
@rh-atomic-bot r+ b9b3abd

**Pull request closed without merging**